protocol-units/settlement/mcr/contracts/script/MultisigMOVETokenDeployer.s.sol

```diff
@@ -0,0 +1,152 @@
+
+import "forge-std/Script.sol";
+import {MOVEToken} from "../src/token/MOVEToken.sol";
+import {TransparentUpgradeableProxy} from "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {Helper, Safe} from "./helpers/Helper.sol";
+import {Vm} from "forge-std/Vm.sol";
+import {ICREATE3Factory} from "./helpers/Create3/ICREATE3Factory.sol";
+import {Enum} from "@safe-smart-account/contracts/common/Enum.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+
+// Script intended to be used for deploying the MOVE token from an EOA
+// Utilizies existing safes and sets them as proposers and executors.
+// The MOVEToken contract takes in the Movement Foundation address and sets it as its own admin for future upgrades.
+// The whole supply is minted to the Movement Foundation Safe.
+// The script also verifies that the token has the correct balances, decimals and permissions.
+contract MultisigMOVETokenDeployer is Helper {
+    using stdJson for string;
+    // COMMANDS
+    // mainnet
+    // forge script MultisigMOVETokenDeployer --fork-url https://eth.llamarpc.com --verify --etherscan-api-key ETHERSCAN_API_KEY
+    // testnet
+    // forge script MultisigMOVETokenDeployer --fork-url https://eth-sepolia.api.onfinality.io/public
+    // Safes should be already deployed
+
+    bytes32 public salt = 0x6c0000000000000000000000018eddf77afc0a5c6d05a564a44fe37b068922c3;
+    bytes32 public constant DEFAULT_ADMIN_ROLE = 0x00;
+
+    function run() external virtual {
+        // load config and deployments data
+        _loadExternalData();
+
+        uint256 signer = vm.envUint("PRIVATE_KEY");
+        vm.startBroadcast(signer);
+
+        // Deploy CREATE3Factory, Safes and Timelock if not deployed
+        _deployDependencies();
+
+        // This deployer solely deploys a timelock and an implementation, it leaves to multisig to execute the deployment
+        // of the actual token.
+        _proposeMultisigMove();
+
+        vm.stopBroadcast();
+
+        if (vm.isContext(VmSafe.ForgeContext.ScriptBroadcast)) {
+            _writeDeployments();
+        }
+    }
+
+    // •☽────✧˖°˖DANGER ZONE˖°˖✧────☾•
+
+    function _proposeMultisigMove() internal {
+        console.log("MOVE: deploying");
+        MOVEToken moveImplementation = new MOVEToken();
+        // genetares bytecode for CREATE3 deployment
+        bytes memory create3Bytecode = abi.encodePacked(
+            type(TransparentUpgradeableProxy).creationCode,
+            abi.encode(
+                address(moveImplementation),
+                address(timelock),
+                abi.encodeWithSignature(moveSignature, deployment.movementFoundationSafe, deployment.movementAnchorage)
+            )
+        );
+
+        deployment.move = create3.getDeployed(deployment.movementDeployerSafe, salt);
+        console.log("MOVE: deployment address", deployment.move);
+        require(_startsWith3073(deployment.move), "MOVE: deployment address does not start with 0x3073");
```

Comment:
Provide note about the security benefits here (this effectively introduces additional work to deploy maliciously).